src/parser/content-parser.php

```diff
@@ -233,10 +233,16 @@ protected function source_block( $block, $registered_blocks, $filter_options ) {
 		$sourced_block = [
 			'name'       => $block_name,
 			'attributes' => $block_attributes,
+			'id'         => wp_unique_id(),
```

Comment:
This was a neat discovery as the IDs are unique, albeit different each time a request is made. The advantage is that we can use this to setup parent-child relationships at least.

Reply:
Do these IDs show up in the REST response as well? If so, can we make it so that we only assign IDs for the GraphQL response? It's not a huge addition of data, but I don't want to add unused metadata to our REST response if it's already in a hierarchical format.

Reply:
Maybe it'd be cleaner to do ID assigning as a post-processing step in `flatten_inner_blocks()`. I think that makes sense, as the parent/child ID relationship is a product of the flattening and GraphQL limitations, not necessarily inherent to the data.

Reply:
>It's not a huge addition of data, but I don't want to add unused metadata to our REST response if it's already in a hierarchical format.
> Maybe it'd be cleaner to do ID assigning as a post-processing step in flatten_inner_blocks().

I should have expanded the reasoning a bit more in my comment.

My main reasoning for keeping this in here, rather than in just the graphQL API was to account for unique IDs playing a role down the line. At the moment, these IDs are generated each time a request is sent. But, what I would like to do is to make the IDs get generated just once and then be able to re-use them. This way, you don't just need to query all the blocks but could actually query a specific block. This could be useful down for a variety of use cases. In addition to that, if the Gutenberg/core folks decide to make block IDs unique we would be able to leverage them here as well.

So this allows us to set that system up. Hope that makes a lot more sense as to why it's done this way?

Reply:
I decided to give this a shot, and I almost have it working. There's repeated outer inner blocks that keep showing up, but otherwise it's working. Once I fix that bug I'll push it up

Reply:
Got it working! It's now only going to be present in the GraphQL, and the side effect of this is that in the filter_options we pass the current id and parent_id of the block being worked on. That should set us up for the future, for the case I mentioned while cleaning up the rest response.